lib/views/home/paymentlist_view.dart

```diff
@@ -53,28 +27,55 @@ class _PaymentListViewState extends State<PaymentListView> {
     return date;
   }
 
+  Future<AppListIncome> listIncome() async {
+    AppListIncome list = await widget.provider.get<AppApi>().listincome();
+
+    /// Sorting the list in decreasing order of time
+    list.incomes.sort((a, b) => a.timeStamp.compareTo(b.timeStamp));
+    List<ListIncome> newList = list.incomes.reversed.toList();
+    list.incomes = newList;
+    return list;
+  }
```

Comment:
Actually the `sort()` function sorts the list in opposite order in which we want it to be so I am reversing it after.

Reply:
Does this mean that you wrote the compere to in the wrong way? Also note that core lighting already keep the list sorted so you need just to reverse it without sorting because core lightning shoulds already do it

Reply:
But this does not exclude that the sort + reverse does not make sense. All these two operations has `>= O(N)` execution time in the worst case so this function is really slow

Reply:
Okay I'll find a different way but lightning does not return the list sorted according to the dates.

Reply:
Okay reversing the order of b and a did the trick.

Reply:
ok nice, remember to move this inside a AppAPI and return the list already sorted inside the UI

Reply:
Done